library/std/src/sys/windows/c.rs

```diff
@@ -1250,19 +1250,15 @@ compat_fn_with_fallback! {
     }
 }
 
-compat_fn_optional! {
+compat_fn_with_fallback! {
     pub static SYNCH_API: &CStr = ansi_str!("api-ms-win-core-synch-l1-2-0");
-
-    // >= Windows 8 / Server 2012
-    // https://docs.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-waitonaddress
-    pub fn WaitOnAddress(
-        Address: LPVOID,
-        CompareAddress: LPVOID,
-        AddressSize: SIZE_T,
-        dwMilliseconds: DWORD
-    ) -> BOOL;
-    pub fn WakeByAddressSingle(Address: LPVOID) -> ();
+    #[allow(unused)]
+    fn WakeByAddressSingle(Address: LPVOID) -> () {
+        crate::sys::windows::thread_parker::unpark_keyed_event(Address)
```

Comment:
This fallback is really dubious if we ever try to use this outside the thread parker. It needs a comment explaining it (and the coupling between this and the parker), although one that just redirects to another comment in the parker could be enough).

(Thanks for exporting it as `wake_by_address_single_or_unpark_keyed_event` rather than just `WakeByAddressSingle::call` tho -- that should at least give people reason to guess that something may be up)

Anyway, ideally this would be less of a leaky abstraction between this and the thread parker. It would be nice to clean this up, but is more important if we start using the windows futex apis anywhere else.

Reply:
Yeah, I did wonder if this was a bit too "clever" hence the attempt at renaming.

Reply:
Ok, for now I've added some comments and a `FIXME` to call this out more explicitly. I can rework it further though.

Reply:
A `FIXME` seems fine for now as it works and is signposted.

That said, we shouldn't leave it indefinitely... it's rather gross for there to be a call into the thread parker's internals from within c.rs, which is otherwise mostly just FFI decls and the like.